### PR TITLE
update static list and open train for some subgraphs

### PR DESCRIPTION
**Pull request by xiaoyao0115**
### PR Category

CINN

### PR Types

Not User Facing

### Description

update static list and open train for some subgraphs

### CODE REVIEW

File: test/ir/pir/cinn/sub_graphs/base.py [hunk 1]
```diff
@@ -96,17 +96,29 @@ def test_ast_prim_cinn(self):
                     st.numpy(), cinn.numpy(), atol=self.atol
                 )
         if self.with_train:
-            st_loss = st_out.mean()
+            if type(st_out) == tuple:
```
Comment (Aurelius84) [inline test/ir/pir/cinn/sub_graphs/base.py:99]: 类型判断可以用isinstance

File: test/ir/pir/cinn/sub_graphs/base.py [hunk 2]
```diff
@@ -96,17 +96,29 @@ def test_ast_prim_cinn(self):
...
+                st_loss, cinn_loss = 0, 0
+                for i in range(len(st_out)):
+                    st_loss += st_out[i].mean()
+                    cinn_loss += cinn_out[i].mean()
+            else:
+                st_loss = st_out.mean()
```
Comment (Aurelius84) [inline test/ir/pir/cinn/sub_graphs/base.py:105]: 计算loss的逻辑可以单独抽离一个类似calcu_loss的函数，组织形式上会更好一些

File: test/ir/pir/cinn/sub_graphs/base.py [hunk 3]
```diff
@@ -96,17 +96,29 @@ def test_ast_prim_cinn(self):
...
+                cinn_loss = cinn_out.mean()
             st_loss.backward()
             st_grad = []
             for i in range(len(st_inputs)):
-                if st_inputs[i].dtype != paddle.int64:
+                if (
+                    st_inputs[i].dtype != paddle.int64
```
Comment (Aurelius84) [inline test/ir/pir/cinn/sub_graphs/base.py:111]: paddle.int32会考虑处理么？

**Review by Aurelius84 — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**